core/commons/che-core-commons-mail/src/main/java/org/eclipse/che/mail/MailSessionProvider.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.mail;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import javax.mail.Authenticator;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import org.eclipse.che.inject.ConfigurationProperties;
+
+/**
+ * Provider of {@link Session} Configuration can be injected from container with help of {@link
```

Comment:
This comment should be put to the appropriate constructor instead. This explanation is not related to the second constructor